src/script/interpreter.cpp

```diff
@@ -2092,6 +2091,105 @@ bool VerifyTaprootCommitment(uint256 &tapleaf_hash,
     return q == q_expected;
 }
 
+#include <logging.h>
+#include <util/strencodings.h>
+
+static bool VerifyScriptType(std::vector<valtype> stack,
+                             const CScript &script_pubkey, uint32_t flags,
+                             const BaseSignatureChecker &checker,
+                             ScriptExecutionMetrics &metrics_out,
+                             ScriptError *serror) {
+    if (script_pubkey.size() == 1 ||
```

Comment:
What makes this a malformed pubkey script?

Reply:
We should put something around this code to be similar:
```
bool CScript::IsPayToScriptHash() const {
    // Extra-fast test for pay-to-script-hash CScripts:
    return (this->size() == 23 && (*this)[0] == OP_HASH160 &&
            (*this)[1] == 0x14 && (*this)[22] == OP_EQUAL);
}
```

And then check `!script_pubkey.IsPayToTapRoot`

Reply:
You also can't really exit early here in the future. Imagine you later have `!script_pubkey.IsPayToTapRoot && !script_pubkey.IsSomeOtherScriptType` then later you still need:

```
if(...IsPayToTapRoot) {
   dostuff
   return 
}

if(...IsSomeOtherScriptType) {
   dostuff
   return 
}
```